src/Analyser/ScopeFactory.php

```diff
@@ -2,17 +2,51 @@
 
 namespace PHPStan\Analyser;
 
+use PhpParser\Node\Expr\Variable;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\StringType;
+
 /** @api */
 class ScopeFactory
 {
 
-	public function __construct(private InternalScopeFactory $internalScopeFactory)
+	public function __construct(
+		private InternalScopeFactory $internalScopeFactory,
+		private bool $explicitMixedForGlobalVariables,
+	)
 	{
 	}
 
 	public function create(ScopeContext $context): MutatingScope
 	{
-		return $this->internalScopeFactory->create($context);
+		$superglobalType = new ArrayType(new StringType(), new MixedType($this->explicitMixedForGlobalVariables));
```

Comment:
as you are working on super global types: psalm recently made them more precise. maybe thats something which we could also think about

https://github.com/vimeo/psalm/blob/7869cb548a5d88399f971ce9f897fe83abbd4213/src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/VariableFetchAnalyzer.php#L594-L813

Reply:
Check the description 😊 I found your issue by luck

Reply:
Ha awesome 😎